tests/web/QueueManager.cpp

```diff
@@ -0,0 +1,169 @@
+//  This file is part of Empirical, https://github.com/devosoft/Empirical
+//  Copyright (C) Michigan State University, 2020.
+//  Released under the MIT Software license; see doc/LICENSE
+
+#include <functional>
+#include <unordered_map>
+
+#include "emp/base/assert.hpp"
+#include "emp/web/_MochaTestRunner.hpp"
+#include "emp/web/Document.hpp"
+#include "emp/prefab/QueueManager.hpp"
+#include "emp/config/SettingConfig.hpp"
+#include "emp/web/web.hpp"
+
+
+struct Test_QueueManager : public emp::web::BaseTest {
+
+  std::function<emp::SettingConfig()> get_setting_config = [](){
+      emp::SettingConfig my_settings;
+      my_settings.AddSetting<int>("my_param") = {9};
+      return my_settings;
+  };
+
+  std::function<int()> get_epochs = [](){return 50;};
+  std::function<std::string()> metric_fun = [](){return "Hello";};
+
+  emp::SettingConfig config;
+  emp::Ptr<emp::QueueManager> queue_manager;
+
+  Test_QueueManager()
+  : BaseTest({"emp_test_container"}) // we can tell BaseTest that we want to create a set of emp::web::Document
+                                     // objects for each given html element ids.
+  {
+    
+    config.AddSetting<int>("my_param") = {6};
+    queue_manager.New(config);
+
+    queue_manager->AddMetric(metric_fun, "Hello metric");
+
+    queue_manager->AddQueueButton(get_setting_config, get_epochs); // Add button and text input
+    queue_manager->BuildTable("my_table"); // Add progress table
+    
+    Doc("emp_test_container")
+      << queue_manager->GetDiv();
+
+    // Test adding metric after table built
+    queue_manager->AddMetric(metric_fun, "Late Hello");
+
+    queue_manager->AddRun(get_setting_config(), get_epochs());
+    queue_manager->AddNewQueuedRunToTable();
+
+    emp_assert(!queue_manager->IsEmpty());
```

Comment:
Should we make these tests `emp_always_assert` so we don't have to worry about them getting turned off if we define NDEBUG?